contrib/epee/src/readline_buffer.cpp

```diff
@@ -196,15 +200,50 @@ static int startup_hook()
   return 0;
 }
 
+static char* completion_matches(const char* text, int state)
+{
+  static size_t list_index;
+  static size_t len;
+
+  if(state == 0)
+  {
+    list_index = 0;
+    len = strlen(text);
+  }
+
+  const std::vector<std::string>& completions = rdln::readline_buffer::get_completions();
+  for(; list_index<completions.size(); )
+  {
+    const std::string& cmd = completions[list_index++];
+    if(cmd.compare(0, len, text) == 0)
```

Comment:
Yes, I know. The static `len` is not strictly needed because it can be computed each invocation or `std::string::find` can be used as I described above to eliminate it. This does require more ops in both cases - I initially forgot that `std::string::find` would have to scan the entire string.